src/query/api/v1/handler/prometheus/native/read.go

```diff
@@ -118,129 +109,130 @@ type RespError struct {
 func NewPromReadHandler(opts options.HandlerOptions) *PromReadHandler {
 	taggedScope := opts.InstrumentOpts().MetricsScope().
 		Tagged(map[string]string{"handler": "native-read"})
-	limits := opts.Config().Limits
 
 	h := &PromReadHandler{
-		engine:              opts.Engine(),
-		fetchOptionsBuilder: opts.FetchOptionsBuilder(),
-		tagOpts:             opts.TagOptions(),
-		limitsCfg:           &limits,
-		promReadMetrics:     newPromReadMetrics(taggedScope),
-		timeoutOps:          opts.TimeoutOpts(),
-		keepEmpty:           opts.Config().ResultOptions.KeepNans,
-		instrumentOpts:      opts.InstrumentOpts(),
+		promReadMetrics: newPromReadMetrics(taggedScope),
+		// timeoutOps:          opts.TimeoutOpts(),
+		// keepEmpty:           opts.Config().ResultOptions.KeepNans,
+		// instrumentOpts:      opts.InstrumentOpts(),
+		opts: opts,
 	}
 
-	pointCount := float64(limits.MaxComputedDatapoints())
-	h.promReadMetrics.maxDatapoints.Update(pointCount)
+	maxDatapoints := opts.Config().Limits.MaxComputedDatapoints()
+	h.promReadMetrics.maxDatapoints.Update(float64(maxDatapoints))
 	return h
 }
 
 func (h *PromReadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	timer := h.promReadMetrics.fetchTimerSuccess.Start()
-	fetchOpts, rErr := h.fetchOptionsBuilder.NewFetchOptions(r)
+	defer timer.Stop()
+
+	ctx := context.WithValue(r.Context(), handler.HeaderKey, r.Header)
+	logger := logging.WithContext(ctx, h.opts.InstrumentOpts())
+
+	ParsedOptions, rErr := ParseRequest(ctx, r, h.opts)
```

Comment:
This is a local var, so can just be lower camel cased yeah? i.e. `parsedOpts`

Reply:
Oops, must have gotten dragged into a rename